### PR TITLE
Refactored Base.search into get_by_most_commom_property

### LINKED ISSUE #3066: Base().search method should support search queries
in current design, `Base.search()` method only accepts the 'value' to be searched and the search query is hardcoded to `{0} = {1}` where {0} is a search key attached to the entity object.
With the `scoped search` feature, we start to support more complex [queries](https://github.com/wvanbergen/scoped_search#query-language), with a number of different operators.

I could create a standalone method to submit the raw queries, but that would make code quite redundant. Therefore I think we might want to extend the current `search()` method.

One of the ways would be to move whole logic (navigating to the page, locating the search form, sending the keys to the search field) to the generic `search_raw` method.
and call it from inside the `search` method, which is just a specific case of the query.

### LINKED ISSUE #3862: Automate positive ui MyAccount Tests

### PR DESCRIPTION
**Pull request by renzon**
close #3066

Created _search returning list of webelements
Created _get returning a single weblement

Methods to be used by subclasses to define search_by*
and get_by* methods

### CODE REVIEW

**Comment by renzon** [comment]
@omaciel @abalakh @elyezer @rplevka 

This PR is probably full of pep8 erros and stuff. I am openning it to check the name get_most_common_property name before fixing it up: https://github.com/SatelliteQE/robottelo/pull/3904/files#diff-6ca60b613b8d4fa334713d53bfd172f3R123

This change would have less impact on code, once it is only a method's name change.

Other approach would be creating properly get methods. E.g.: get_by_login for User, get_by_id for Errata and so on. But this would lead to some work on tests code themselves, but would be more explicit.
Also:
search_key could be removed and subclasses could override the get_most_common_property. Is it better?

What you guys think?

**Comment by omaciel** [comment]
@renzon how about a hands on demo/discussion tomorrow?

**Comment by renzon** [comment]
@omaciel: In fact I thought exactly this whe  creating this PR. Tomorrow I explain that and some other issues around this ;)

**Comment by oshtaier** [comment]
@renzon yeah, because we made a decision and had a discussions about Base.search around whole team. Probably, it was better to have some meetings before that implementation.
Anyway, let's have some demo tomorrow

**Comment by renzon** [comment]
After meeting we decide the name "get_entity" to replace the old "search" now and discuss architecture enhancement on #3720. Here are the test results this PR:
```console
============================= 368 tests deselected =============================
 91 failed, 582 passed, 31 skipped, 368 deselected, 14 error in 69095.68 seconds
```

errors seems not to be related with this PR. So I am going to fix pep8 and send email to all team to spread the change.

**Comment by rochacbruno** [comment]
I guess you mean #3906

File: robottelo/ui/base.py [hunk 1]
```diff
@@ -113,24 +115,57 @@ def navigate_to_entity(self):
...
+        return self.click(self.get_entity(element))
 
-    def search(self, element):
+    def get_entity(self, element):
+        """Uses the search box to locate an element from a list of elements by
+        its most common property.
+        The most common property is defined as 'search_key' entity class
+        attribute. If not defined, property name 'name' is used.
+
+        :param element: either element name or a tuple, containing element name
```
Comment (omaciel) [inline robottelo/ui/base.py]: Can you add examples for both possible ways `element` can be used?

File: robottelo/ui/base.py [hunk 2]
```diff
@@ -343,8 +379,8 @@ def wait_until_element(self, locator, timeout=12, poll_frequency=0.5):
             )
             return None
 
-    def wait_until_element_is_clickable(
-            self, locator, timeout=12, poll_frequency=0.5):
+    def wait_until_element_is_clickable(self, locator, timeout=12,
```
Comment (omaciel) [inline robottelo/ui/base.py]: Is this necessary? It seems to go against the way we have been formatting functions' arguments throughout the code.

File: robottelo/ui/base.py [hunk 3]
```diff
@@ -370,8 +406,8 @@ def wait_until_element_is_clickable(
             )
             return None
 
-    def wait_until_element_is_not_visible(
-            self, locator, timeout=12, poll_frequency=0.5):
+    def wait_until_element_is_not_visible(self, locator, timeout=12,
+                                          poll_frequency=0.5):
```
Comment (omaciel) [inline robottelo/ui/base.py]: Is this necessary? It seems to go against the way we have been formatting functions' arguments throughout the code.

**Comment by omaciel** [comment]
@oshtaier @abalakh @svtkachenko could I ask you guys to review this?

**Comment by omaciel** [comment]
@lpramuk you too :)

**Comment by rochacbruno** [comment]
@omaciel I understood that we are not going to merge this, and wait to discuss more about some other approaches for this refactor. We already started the discussion here: https://github.com/SatelliteQE/robottelo/issues/3906

**Comment by abalakh** [comment]
yep, that's what we've discussed on friday

**Comment by omaciel** [comment]
Thanks guys, I have added my comment on #3906

**Comment by renzon** [comment]
@rochabruno, @abalakh: I thought we would do this minor change know to allow search on other properties. But once I was wrong, I am closing this issue.

**Pull request closed without merging**